ui/apps/platform/src/Containers/MainPage/InviteUsersModal.tsx

```diff
@@ -1,21 +1,126 @@
+                        .split(',')
+                        .map((v) => v.trim())
+                        .every(isEmail)
+            )
+        ), // .string().trim().required('At least one email is required'),
+    provider: yup.string().required('An auth provider is required'),
+    role: yup.string().required('A role is required'),
+});
+
+const defaultInviteFormValues: InviteFormValues = {
+    emails: '',
+    provider: '',
+    role: '',
+};
 
 const feedbackState = createStructuredSelector({
+    authProviders: selectors.getAvailableAuthProviders,
+    groups: selectors.getRuleGroups,
+    roles: selectors.getRoles,
+    isFetchingAuthProviders: (state) =>
+        selectors.getLoadingStatus(state, authActionTypes.FETCH_AUTH_PROVIDERS) as boolean,
+    isFetchingRoles: (state) =>
+        selectors.getLoadingStatus(state, roleActionTypes.FETCH_ROLES) as boolean,
     invite: selectors.inviteSelector,
 });
 
 function InviteUsersModal(): ReactElement | null {
     const { invite: showInviteModal } = useSelector(feedbackState);
     const dispatch = useDispatch();
+    const formik = useFormik<InviteFormValues>({
+        initialValues: defaultInviteFormValues,
+        onSubmit: () => {}, // required but not used, because the submit action is in the modal footer
+        validationSchema,
+        validateOnMount: true,
+    });
+
+    const { isValid, values, setFieldValue } = formik;
+
+    const { authProviders, groups, roles } = useSelector(feedbackState);
+
+    const authProvidersWithRules = mergeGroupsWithAuthProviders(authProviders, groups);
+
+    const allowedRoles = roles.filter((role) => {
+        return role.name !== 'Admin' && role.name !== 'None';
+    });
+
+    useEffect(() => {
+        dispatch(authActions.fetchAuthProviders.request());
+        dispatch(roleActions.fetchRoles.request());
+        dispatch(groupActions.fetchGroups.request());
+    }, [dispatch]);
+
+    useEffect(() => {
+        const typedProviders = authProviders as AuthProvider[]; // redux state from sagas is not yet typed
+
+        // if there is only 1 authProvider, pre-select it in the dropdown
+        if (typedProviders.length === 1) {
+            // eslint-disable-next-line no-void
+            void setFieldValue('provider', typedProviders[0].name);
+        } else if (typedProviders.length > 1) {
+            // if there is more than 1 authProvider, pre-select RedHat / OpenShift provider if possible
+            const redhatSsoProvider = typedProviders.find(
+                (provider) =>
+                    provider.name.includes('Red Hat SSO') || provider.name.includes('Red Hat SSO')
+            );
+
+            if (redhatSsoProvider) {
+                // eslint-disable-next-line no-void
+                void setFieldValue('provider', redhatSsoProvider.name);
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [authProviders]); // ignore `setFieldValue` as a dependency
```

Comment:
Was `setFieldValue` causing a problem when added to the dependency array? It seems like that should be a stable value from the formik hook and shouldn't cause a re-render loop.

Reply:
It was causing almost constant re-rendering, when it was part of the dependency array.

Reply:
To ask similar question from a different angle, in the theoretical possibility of a change, would `setFieldValue` overwrite a user selection in the form?

Is there a different way to initialize the values than via imperative calls and double `useEffect` hooks?

For example, as mentioned above, if via service function calls instead of `dispatch` actions in `useEffect` the can the same hook call pure functions factored outside modal but in its file to do one batch update to formik?

Reply:
Yeah, the groupsbatch update might need to use redux and sagas to make sure that auth provider form displays consistent information. It does not dispatch a request to refresh data as far as I can see.

Reply:
I am reluctant to modify or expand the way we handle API calls for auth providers just for this minor feature.